pkg/kubectl/cmd/util/factory.go

```diff
@@ -354,6 +354,19 @@ func NewFactory(optionalClientConfig clientcmd.ClientConfig) *Factory {
 					return nil, errors.New("provided options object is not a PodLogOptions")
 				}
 				return c.Pods(t.Namespace).GetLogs(t.Name, opts), nil
+
+			case *api.ReplicationController:
+				opts, ok := options.(*api.PodLogOptions)
+				if !ok {
+					return nil, errors.New("provided options object is not a PodLogOptions")
+				}
+				selector := labels.SelectorFromSet(t.Spec.Selector)
+				pod, err := GetFirstPod(c, t.Namespace, selector)
```

Comment:
I think this should return an error if the rc has more than 1 pod. I don't know that there are any guarantees about ordering when listing pods with a selector; if there aren't any, rerunning the same command on a multi-pod rc might pick a different "first pod".